server/app/models/sync.rb

```diff
@@ -147,10 +147,12 @@ def perform_post_discard_sync
   def stream_name_exists?
     return if destination.blank?
 
-    stream = destination.catalog.find_stream_by_name(stream_name)
-    return if stream.present?
-
-    errors.add(:stream_name,
-               "Add a valid stream_name associated with destination connector")
+    stream = destination || destination.catalog&.find_stream_by_name(stream_name)
+    if stream.catalog.blank?
+      errors.add(:catalog, "Catalog is missing")
+    elsif stream.catalog.find_stream_by_name(stream_name).blank?
+      errors.add(:stream_name,
+                 "Add a valid stream_name associated with destination connector")
```

Comment:
need a spec for the stream_name validation as well

Reply:
This is test in spec at line 170-179.